File: single-language-analysis/dataset-files/java/TestSmell091.java
import org.junit.Test;
import static org.junit.Assert.*;
import java.io.File;

public class TestSmell091 {

    @Test
    public void testMakedirs() {
        Path d = new Path(tmpdir);

        File tempf = new File(d.toString(), "temp.txt");
        try {
            tempf.createNewFile();
            Path foo = new Path(d.toString(), "foo");
            Path boz = new Path(foo.toString(), "bar", "baz", "boz");
            boz.makedirs();
            try {
                assertTrue(boz.isdir());
            } finally {
                boz.removedirs();
            }
            assertFalse(foo.exists());
            assertTrue(d.exists());

            foo.mkdir(0750);
            boz.makedirs(0700);
            try {
                assertTrue(boz.isdir());
            } finally {
                boz.removedirs();
            }
            assertFalse(foo.exists());
            assertTrue(d.exists());
        } finally {
            tempf.delete();
        }
    }
}

File: single-language-analysis/dataset-files/python/test_smell_091.py
# Original URL: https://github.com/jvaldiviezo9/path.py/blob/3684c4dc4896c174852b901b033e021a6267df2e/test_path.py#L450-L477

class TestScratchDir:

    def test_smell_091(self, tmpdir):
        d = Path(tmpdir)

        # Placeholder file so that when removedirs() is called,
        # it doesn't remove the temporary directory itself.
        tempf = d / 'temp.txt'
        tempf.touch()
        try:
            foo = d / 'foo'
            boz = foo / 'bar' / 'baz' / 'boz'
            boz.makedirs()
            try:
                assert boz.isdir()
            finally:
                boz.removedirs()
            assert not foo.exists()
            assert d.exists()

            foo.mkdir(0o750)
            boz.makedirs(0o700)
            try:
                assert boz.isdir()
            finally:
                boz.removedirs()
            assert not foo.exists()
            assert d.exists()
        finally:
            os.remove(tempf)